tests/functional/settings_change_email.js

```diff
@@ -0,0 +1,144 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+define([
+  'intern',
+  'intern!object',
+  'tests/lib/helpers',
+  'tests/functional/lib/helpers',
+  'tests/functional/lib/selectors',
+], function (intern, registerSuite, TestHelpers, FunctionalHelpers, selectors) {
+
+  const config = intern.config;
+  const SIGNUP_URL = config.fxaContentRoot + 'signup';
+  const PASSWORD = 'password';
+  const NEW_PASSWORD = 'password1';
+
+  let email;
+  let secondaryEmail;
+
+  const clearBrowserState = FunctionalHelpers.clearBrowserState;
+  const click = FunctionalHelpers.click;
+  const fillOutChangePassword = FunctionalHelpers.fillOutChangePassword;
+  const fillOutResetPassword = FunctionalHelpers.fillOutResetPassword;
+  const fillOutCompleteResetPassword = FunctionalHelpers.fillOutCompleteResetPassword;
+  const fillOutSignIn = FunctionalHelpers.fillOutSignIn;
+  const fillOutSignUp = FunctionalHelpers.fillOutSignUp;
+  const openPage = FunctionalHelpers.openPage;
+  const openVerificationLinkInNewTab = FunctionalHelpers.openVerificationLinkInNewTab;
+  const openVerificationLinkInSameTab = FunctionalHelpers.openVerificationLinkInSameTab;
+  const testIsBrowserNotified = FunctionalHelpers.testIsBrowserNotified;
+  const testElementExists = FunctionalHelpers.testElementExists;
+  const testElementTextEquals = FunctionalHelpers.testElementTextEquals;
+  const testErrorTextInclude = FunctionalHelpers.testErrorTextInclude;
+  const testSuccessWasShown = FunctionalHelpers.testSuccessWasShown;
+  const type = FunctionalHelpers.type;
+  const visibleByQSA = FunctionalHelpers.visibleByQSA;
+
+  registerSuite({
+    name: 'settings change email',
+
+    beforeEach: function () {
+      email = TestHelpers.createEmail();
+      secondaryEmail = TestHelpers.createEmail();
+      return this.remote.then(clearBrowserState())
+        .then(openPage(SIGNUP_URL, selectors.SIGNUP.HEADER))
+        .then(fillOutSignUp(email, PASSWORD))
+        .then(testElementExists('#fxa-confirm-header'))
+        .then(openVerificationLinkInSameTab(email, 0))
+        .then(testElementExists(selectors.SETTINGS.HEADER))
+        .then(click('#emails .settings-unit-stub button'))
+
+        // add secondary email, resend and remove
```

Comment:
This `beforeEach` function seems like it's doing too much, esp the `resend and remove` section, which seems like it should be its own test.